nss_cache/sources/ldapsource.py

```diff
@@ -397,9 +397,15 @@ def __iter__(self):
         # If the dn is requested, return it along with the payload,
         # otherwise ignore it.
         for key in record[1]:
-          if isinstance(record[1][key][0], bytes) and key != 'objectSid':
-            value = record[1][key][0].decode('utf-8')
-            record[1][key] = [value]
+          if len(record[1][key]) > 1:
+            for i, item in enumerate(record[1][key]):
+              if isinstance(item, bytes) and key != 'objectSid':
+                value = item.decode('utf-8')
+                record[1][key][i] = value
+          else:
+            if isinstance(record[1][key][0], bytes) and key != 'objectSid':
+              value = record[1][key][0].decode('utf-8')
+              record[1][key] = [value]
```

Comment:
and all of this doesn't need to exist, and instead finally set record[1][key] = tmp_list.

Reply:
IMHO I think the only way to make it work for the tests as well, of course it will work with a tmp_list:
```
      for record in data:
        # If the dn is requested, return it along with the payload,
        # otherwise ignore it.
        for key in record[1]:
          for i, item in enumerate(record[1][key]):
            if isinstance(item, bytes) and key != 'objectSid':
              value = item.decode('utf-8')
              record[1][key][i] = value
        if self._dn_requested:
          merged_records = {'dn': record[0]}
          merged_records.update(record[1])
          yield merged_records
        else:
          yield record[1]
```

Reply:
This is a possible solution as well:
```
        for key in record[1]:
          for i in range(len(record[1][key])):
            if isinstance(record[1][key][i], bytes) and key != 'objectSid':
              value = record[1][key][i].decode('utf-8')
              record[1][key][i] = value
```